doc/third-party-go-clients.md

```diff
@@ -1,15 +1,20 @@
-# Third party Go clients
+# How to create and use go clients for Juju
 
-This describes an approach to allow third party Go codebases to drive Juju.  At
-a high level it works like this.
+This document shows how to create a Golang client to drive Juju.
 
-1. Create a connector instance, telling it how to connect to a Juju controller
-   and how to provide authentication credentials.
-2. Use this connector instance to obtain a connection instance.
-3. Get a facade client from the connection.
-4. Use the facade client to make one or more requests.
+Steps: 
```

Comment:
If you want the steps to function as a TOC for the doc (imo a good pattern indeed), let's just call this "Contents:".